autoload/gh/issues.vim

```diff
@@ -271,32 +272,75 @@ function! s:create_issue_success(resp) abort
   call gh#gh#message(printf('new issue: %s', a:resp.body.html_url))
 endfunction
 
-function! s:set_issue_template_buffer(resp) abort
+function! s:set_issue_title(resp) abort
   let gh_issue_title = input('[gh.vim] issue title ')
   echom ''
   redraw
   if gh_issue_title is# ''
     call gh#gh#error_message('no issue title')
     return
   endif
+  let b:gh_issue_new['title'] = gh_issue_title
+  let b:gh_issue_new['issue_template_list'] = a:resp.body
+  call s:get_assignees_list()
+endfunction
+
+function! s:get_assignees_list() abort
+  call gh#github#repos#get_assignees(b:gh_issue_new.owner, b:gh_issue_new.name)
+        \.then(function('s:open_assignees_list'))
+        \.catch({err -> execute('%d_ | call gh#gh#set_message_buf(err.body)', '')})
+endfunction
+
+function! s:open_assignees_list(resp) abort
+  let s:issue_new_assignees = ["-"]
```

Comment:
What is the intent of this?

Reply:
@skanehira 
I intended to express for 'assigning nobody' by this '-'.
if you don't want to add assignee on new issue, not selecting any item by <C-i> in quickpick-buffer is not enough for assigning nobody. because the item which is on your cursor would be selected even if you don't selected an items by <C-i>. 

autoload/gh/provider/quickpick.vim in `s:on_accept` function around line 340 is the place.
``` vim
      else
        let l:items = [s:state['fitems'][l:index]] " <= l:index is the cursor line
      endif
```

so, I placed '-' on top of the list.
then, when you not selecting <C-i> in quickpick-buffer, the item '-' is selected(if you didn't move cursor in quickpick buffer), and `s:on_accept_assignees` function in gh/issues.vim remove that '-' and finally nobody assignees on new issue would be achieved.

this is the reason why i set '-' as default item.

Reply:
> I intended to express for 'assigning nobody' by this '-'.
> if you don't want to add assignee on new issue, not selecting any item by in quickpick-buffer is not enough for assigning nobody. because the item which is on your cursor would be selected even if you don't selected an items by .

How about implement cancel function?
I think if user do nothing, just close quickpick is better way.
As a result, operation and implement will be more simple because we don't need to remove `-` from list and user is just press `<C-c>` to close quickpick.

Reply:
> How about implement cancel function?
i thought it would be better that the cancel function (<C-c>) should be able to cancel for creating new issue itself, not for selecting nobody as assignee, like below.

1. `vnew gh://:owner/:repo/issues/new`
2. Enter title of new issue.
3. Select assignee(s). ( there should be 3 options for user )
- Pattern A: `<C-c> `
    - => cancel creating new issue 
- Pattern B: Just put `<CR>`  (without selecting)
    - => create new issue, but no assignee
- Pattern C: Select assignee(s) and put `<CR>`
    - => create new issue with assignee(s)

How about [this fix](https://github.com/skanehira/gh.vim/pull/119/commits/2f1f7ac6b4ff8e6365230143cdbeeb50574b8007) ?
I fixed `autoload/gh/provider/quickpick.vim` a little bit.
- removed `-` from assignees list
- user can select no assignee by just putting `<CR>` without selecting nobody

Reply:
> Pattern C: Select assignee(s) and put <CR>
> => create new issue with assignee(s)

I thought it's inconvenience a little bit when we choose how to merge the PR, because we must choose one.
How about implement like `gh issue create`? We can choose option to add some metadata.

![image](https://user-images.githubusercontent.com/7888591/112826184-9ce3d280-90c7-11eb-8303-78a9f0454a62.png)

Reply:
@skanehira 
> I thought it's inconvenience a little bit when we choose how to merge the PR, because we must choose one.

sorry, I didn't consider it. I've got to understand.
I'll rebase the commit :)

> How about implement like `gh issue create` ? We can choose option to add some metadata.

hmm, sorry. could you explain your idea, a bit more...?
implement like the picture (you posted), just before selecting assignees?

Reply:
> implement like the picture (you posted), just before selecting assignees?

In my image, operation is bellow.

1. open create new issue buffer
2. choose template (if existed)
3. edit issue body and save with `:w`
4. ask user they want to add some metadata
  1. if they want, ask they what kind of metadata to add (like screenshot) with quickpick (labels, projects, milestone will implement in other issues)
  2. if they don't want, do nothing
5. create new issue

Reply:
Thank you for your explanation, I've got your image clearly :)
I'm gonna try to implement it.